### PR TITLE
🎨 introducing parent ids to rut (🗃️)

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🎨 introduce project/node parent fields to the resource usage tracker
- 🎨 @odeimaiz I have added `root_parent_project_id` & `root_parent_project_name` to the ServiceRunGet. You can start to use them in the frontend. I have also removed the `service_specifications` dictionary field as it is not used as far as I know. 

## Related issue/s
- relates https://github.com/ITISFoundation/osparc-simcore/issues/5878
- relates https://github.com/ITISFoundation/osparc-simcore/issues/5816

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]
Thanks. As discussed, if I receive a valid ``root_parent_project_name``, it will be shown in the ``project_name``.

File: packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py [hunk 1]
```diff
@@ -117,9 +117,40 @@ class ResourceTrackerServiceRunStatus(str, enum.Enum):
...
+        doc="We want to store the root parent project name for tracking/billing purposes and be sure it stays there even when the node is deleted (that's also reason why we do not introduce foreign key)",
+    ),
+    sa.Column(
+        "parent_node_id",  # UUID
+        sa.String,
+        nullable=False,
+        doc="we want to store the parent node/service id for tracking/billing purposes and be sure it stays there even when the node is deleted.",
+    ),
+    sa.Column(
+        "root_parent_node_id",  # UUID
```
Comment (sanderegg) [inline packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py:149]: so here you do not need the name?

File: services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py [hunk 2]
```diff
@@ -106,6 +106,11 @@ async def publish_service_resource_tracking_started(  # pylint: disable=too-many
         project_name=project_name,
         node_id=node_id,
         node_name=node_name,
+        parent_project_id=project_id,  # <-- SAN please modify
+        root_parent_project_id=project_id,  # <-- SAN please modify
+        root_parent_project_name=project_name,  # <-- SAN please modify
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py:111]: :+1:

**Review by sanderegg — APPROVED** [review]
all good thanks!

**Review by pcrespov — APPROVED** [review]
thx

File: packages/models-library/src/models_library/rabbitmq_messages.py [hunk 3]
```diff
@@ -217,6 +217,13 @@ class RabbitResourceTrackingStartedMessage(RabbitResourceTrackingBaseMessage):
     node_id: NodeID
     node_name: str
 
+    parent_project_id: ProjectID
+    root_parent_project_id: ProjectID
+    root_parent_project_name: str
+
+    parent_node_id: NodeID
+    root_parent_node_id: NodeID
```
Comment (GitHK) [inline packages/models-library/src/models_library/rabbitmq_messages.py:225]: What is the difference between these?

File: packages/models-library/src/models_library/rabbitmq_messages.py [hunk 4]
```diff
@@ -217,6 +217,13 @@ class RabbitResourceTrackingStartedMessage(RabbitResourceTrackingBaseMessage):
     node_id: NodeID
     node_name: str
 
+    parent_project_id: ProjectID
+    root_parent_project_id: ProjectID
```
Comment (GitHK) [inline packages/models-library/src/models_library/rabbitmq_messages.py:221]: And what is the difference between these?

File: packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py (hunk 1 above)
Comment (matusdrobuliak66) [reply packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py:149]: no, I think the root/parent node name will not be used, so I didn't add it for now.

File: packages/models-library/src/models_library/rabbitmq_messages.py (hunk 3 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/rabbitmq_messages.py:225]: Project ID | Parent Project ID | Root Parent Project ID |
--- | --- | --- | 
A | NULL | NULL | 
B | A | A | 
C | B | A | 

As we support via API server to initialize projects inside of projects, the root parent project ID is the project ID from which the user initialized the computation. The same holds for the node inside of the project.

File: packages/models-library/src/models_library/rabbitmq_messages.py (hunk 4 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/rabbitmq_messages.py:221]: The same as above

File: packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py [hunk 5]
```diff
@@ -117,9 +117,40 @@ class ResourceTrackerServiceRunStatus(str, enum.Enum):
     sa.Column(
         "node_name",
         sa.String,
-        nullable=True,
+        nullable=False,
```
Comment (bisgaard-itis) [inline packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py:120]: I like these changes, but should it be checked that this is doable for all the data in the db on the different deployments? Could it be that there is data around which would not allow this?

**Review by bisgaard-itis — APPROVED** [review]
Very nice, thanks a lot. Just a single question (probably more for my understanding)

File: packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py (hunk 5 above)
Comment (matusdrobuliak66) [reply packages/postgres-database/src/simcore_postgres_database/models/resource_tracker_service_runs.py:120]: Yes, I have checked it upfront, they are even defined as non-nullable in the `projects` table.

File: packages/models-library/src/models_library/rabbitmq_messages.py (hunk 3 above)
Comment (GitHK) [reply packages/models-library/src/models_library/rabbitmq_messages.py:225]: Sorry so `parent` and `root_parent` are the same thing? These names are very confusing to me.

**Review by GitHK — APPROVED** [review]
Thanks, this is better now

**Pull request merged**